src/libUtils/ShardSizeCalculator.cpp

```diff
@@ -58,13 +58,13 @@ uint32_t ShardSizeCalculator::CalculateShardSize(const uint32_t numberOfNodes) {
 }
 
 void ShardSizeCalculator::GenerateShardCounts(
-    const uint32_t shardSize, const uint32_t shardSizeThreshold,
-    const uint32_t numNodesForSharding, vector<uint32_t>& shardCounts,
-    bool logDetails) {
+    const uint32_t shardSize, const uint32_t shardSizeThresholdLo,
+    const uint32_t shardSizeThresholdHi, const uint32_t numNodesForSharding,
+    vector<uint32_t>& shardCounts, bool logDetails) {
   LOG_MARKER();
 
-  const uint32_t SHARD_THRESHOLD_LO = shardSize - shardSizeThreshold;
-  const uint32_t SHARD_THRESHOLD_HI = shardSize + shardSizeThreshold;
+  const uint32_t SHARD_THRESHOLD_LO = shardSize - shardSizeThresholdLo;
+  const uint32_t SHARD_THRESHOLD_HI = shardSize + shardSizeThresholdHi;
```

Comment:
Actually the shardSizeThresholdLo is a kind of tolerance and SHARD_THRESHOLD_LO is the threshold value. Suggest to change naming of shardSizeThresholdLo and all relative variables to xxxTolerance, to make the implementation easier to understand.